### PR TITLE
Added touch swipe to side bar

### PR DESCRIPTION
**Pull request by ayush015**

### CODE REVIEW

File: views/footer_imports.ejs [hunk 1]
```diff
@@ -1,4 +1,8 @@
+<script src="https://code.jquery.com/jquery-3.2.1.slim.min.js" integrity="sha384-KJ3o2DKtIkvYIK3UENzmM7KCkRr/rE9/Qpg6aAZGJwFDMVNA/GpGFF93hXpG5KkN" crossorigin="anonymous"></script>
 <script src="https://cdnjs.cloudflare.com/ajax/libs/popper.js/1.12.9/umd/popper.min.js" integrity="sha384-ApNbgh9B+Y1QKtv3Rn7W3mgPxhU9K/ScQsAP7hUibX39j7fakFPskvXusvfa0b4Q" crossorigin="anonymous"></script>
-<script src="https://stackpath.bootstrapcdn.com/bootstrap/4.3.1/js/bootstrap.min.js" integrity="sha384-JjSmVgyd0p3pXB1rRibZUAYoIIy6OrQ6VrjIEaFf/nJGzIxFDsf4x0xIM+B07jRM" crossorigin="anonymous"></script>
+<script src="https://maxcdn.bootstrapcdn.com/bootstrap/4.0.0/js/bootstrap.min.js" integrity="sha384-JZR6Spejh4U02d8jOt6vLEHfe/JQGiRRSQQxSfFWpi1MquVdAyjUar5+76PVCmYl" crossorigin="anonymous"></script>
+<script src='https://cdnjs.cloudflare.com/ajax/libs/jquery/2.1.3/jquery.min.js'></script>
+<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery.touchswipe/1.6.19/jquery.touchSwipe.min.js" integrity="sha512-YYiD5ZhmJ0GCdJvx6Xe6HzHqHvMpJEPomXwPbsgcpMFPW+mQEeVBU6l9n+2Y+naq+CLbujk91vHyN18q6/RSYw==" crossorigin="anonymous"></script>
+<script src='/notify.min.js'></script>
```
Comment (LalitNM) [inline views/footer_imports.ejs]: Check this code again and add only those cdns, which are necessary. Remove all redundant cdns including `notify.min.js` as this is deleted in latest version of this repository.

File: views/head_imports.ejs [hunk 2]
```diff
@@ -4,4 +4,4 @@
 <link rel="stylesheet" href="/emojify.css">
 <script src='https://cdnjs.cloudflare.com/ajax/libs/jquery/3.2.1/jquery.min.js'></script>
 <script src="https://cdnjs.cloudflare.com/ajax/libs/socket.io/1.4.5/socket.io.js"></script>
-<script src="/client.js"></script>
\ No newline at end of file
+<script src="/client.js"></script>
```
Comment (LalitNM) [inline views/head_imports.ejs]: I don't see any change in this file, please do not make unnecessary changes.

File: views/index.ejs [hunk 3]
```diff
@@ -44,8 +44,8 @@
             <i id="toggle-icon" class="fa fa-angle-right"></i>
           </button>
         </div>
-      </div>
-      <div class="right active" id="lobby-msg" data-joined="1">
+      </div> 
+      <div class="swipe-area right active" id="lobby-msg" data-joined="1">
```
Comment (LalitNM) [inline views/index.ejs]: adding `swipe-area` class here adds swipe action in lobby-room, but this action is not working in any room other than lobby. You need to add this in `client.js` too.

**Review by LalitNM — CHANGES_REQUESTED** [review]
In addition to these changes, now the modal doesn't collapses after hitting `Add Room` button. Rectify this also.

Do these changes and then I will again review your pull request.

File: views/index.ejs (hunk 3 above)
Comment (ayush015) [reply views/index.ejs]: I have made the required changes .

File: public/script.js [hunk 4]
```diff
@@ -178,3 +178,15 @@ function notify(data, type){
...
+// swipeable side bar
+$(".swipe-area").swipe({
+  swipeStatus:function(event, phase, direction, distance, duration, fingers)
+    {
+      if (phase=="move" && direction =="right") {
+         $(".left").addClass("open-menu");
+      }
+      if (phase=="move" && direction =="left") {
+         $(".left").removeClass("open-menu");
+      }
```
Comment (LalitNM) [inline public/script.js]: In addition to these, you have to change class of `#toggle-icon` also. You have to change its class from `fa-angle-right` to `fa-angle-left` and vice-versa.

File: public/client.js [hunk 5]
```diff
@@ -80,7 +80,6 @@ const appendUserInfo = (room_name, description) => {
     $('.card-columns').append($userInfo);
 
 }
-
```
Comment (LalitNM) [inline public/client.js]: remove this unnecessary change.

**Review by LalitNM — CHANGES_REQUESTED** [review]
And few more changes you have to do

- remove `package-lock.json` from commit.

- squash all commits in a single commit to make cleaner commit history.

File: public/script.js (hunk 4 above)
Comment (ayush015) [reply public/script.js]: Okay Yeah I'll do that .

File: public/client.js (hunk 5 above)
Comment (ayush015) [reply public/client.js]: That's just a line added without any code, anyways i'll revert it !  😀

**Comment by ayush015** [comment]
All the required changes are made !

**Comment by LalitNM** [comment]
Great! Now only squash left.

**Comment by ayush015** [comment]
> Great! Now only squash left.

I'm getting unnecessary issues while squashing it .Can we just merge it ?  😅

**Comment by LalitNM** [comment]
I'm sure that there is a way to clean this commit history, can you come on gitter I can explain you there. Just drop a message in 'Git' room on gitter. Follow [this](https://gitter.im/Baat-Cheet-IIT) link.

**Comment by ayush015** [comment]
> I'm sure that there is a way to clean this commit history, can you come on gitter I can explain you there. Just drop a message in 'Git' room on gitter. Follow [this](https://gitter.im/Baat-Cheet-IIT) link.

Done !

File: views/index.ejs [hunk 6]
```diff
@@ -45,7 +45,7 @@
           </button>
         </div>
       </div>
-      <div class="swipe-area right active" id="lobby-msg" data-joined="1">
+      <div class="right active" id="lobby-msg" data-joined="1">
```
Comment (LalitNM) [inline views/index.ejs]: No need to remove this swipe-area.

File: views/index.ejs [hunk 7]
```diff
@@ -13,7 +13,7 @@
 <body>
   <%- include('user', {title: title}); %>
   <div class="wrapper">
-    <div class="app-container">
+    <div class="app-container swipe-area">
```
Comment (LalitNM) [inline views/index.ejs]: instead remove this one.

Comment (ayush015) [reply views/index.ejs]: then it wont work with other rooms only lobby would work

Comment (LalitNM) [reply views/index.ejs]: For other room you are adding this class in client.js.

**Comment by LalitNM** [comment]
@ayush015 the problem is still there, but don't worry I will rectify that. I'm a bit busy nowadays.

**Comment by ayush015** [comment]
> @ayush015 the problem is still there, but don't worry I will rectify that. I'm a bit busy nowadays.

Okay its totally cool .So should I close this PR?

**Comment by LalitNM** [comment]
No, leave it , let me approve your changes.

**Review by LalitNM — APPROVED** [review]

**Comment by ayush015** [comment]
> No, leave it , let me approve your changes.

I have resolved the merge conflict Should i push it again?
In the meanwhile i'll also look for the solution !

**Comment by LalitNM** [comment]
I resolved them for now. You can search for some solution and if you find it, then you should push it.

**Comment by ayush015** [comment]
> I resolved them for now. You can search for some solution and if you find it, then you should push it.

Yeah totally cool .I'll try find something and then i'll push it

**Comment by ayush015** [comment]
> > I resolved them for now. You can search for some solution and if you find it, then you should push it.
> 
> Yeah totally cool .I'll try find something and then i'll push it

**Comment by ayush015** [comment]
@LalitNM Hey man I have found a way we can resolve this . Just need little of your help !
Just want to be clear on the issue! I'm writing the issue below just let me know if this is the issue or its something else

So while we start the app the text area is not usable we cannot click and are not able to type anything right ?

**Comment by LalitNM** [comment]
@ayush015 Yes, we cannot focus it by clicking on it. Also try to push the send button, this will focus the textarea, because the function behind that button focuses the textarea.

**Comment by ayush015** [comment]
> @ayush015 Yes, we cannot focus it by clicking on it. Also try to push the send button, this will focus the textarea, because the function behind that button focuses the textarea.

I have found a different approach . Its working !

**Comment by LalitNM** [comment]
@ayush015, I was just wondering that why you added a cdn just for one swipe, I don't think we will be using another function from that library. I want something similar to [this](https://stackoverflow.com/questions/15084675/how-to-implement-swipe-gestures-for-mobile-devices). And please rebase your branch so that you don't have to resolve conflicts, or you can make another pull request.

**Comment by ayush015** [comment]
> @ayush015, I was just wondering that why you added a cdn just for one swipe, I don't think we will be using another function from that library. I want something similar to [this](https://stackoverflow.com/questions/15084675/how-to-implement-swipe-gestures-for-mobile-devices). And please rebase your branch so that you don't have to resolve conflicts, or you can make another pull request.

hey @LalitNM i'm busy with some other work right now . I'll get to it later if that's fine with you !

**Comment by LalitNM** [comment]
> hey @LalitNM i'm busy with some other work right now . I'll get to it later if that's fine with you !

No need to worry about this @ayush015, I did required changes and merged this pull request. You can look at some other issue.
By the way, it was great what you did :clap:
